File: 7kyu/Deodorant Evaporator/evaporator.ts
// https://www.codewars.com/kata/5506b230a11c0aeab3000c1f/typescript

export function evaporator(
  content: number,
  evapPerDay: number,
  threshold: number
): number {
  // your code
  let rest = content;
  const thresholdInMl = (content * threshold) / 100;
  let nth = 1;

  while (true) {
    rest -= (rest * evapPerDay) / 100;

    if (rest < thresholdInMl) {
      break;
    }

    nth += 1;
  }

  return nth;
}

// https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/log
// log를 활용해 푸는 접근법이 정말 신선하다
// export function evaporator(
//   content: number,
//   evap_per_day: number,
//   threshold: number
// ): number {
//   return Math.ceil(
//     Math.log(threshold / 100) / Math.log(1 - evap_per_day / 100)
//   );
// }

File: 7kyu/Deodorant Evaporator/Evaporator.kt
// https://www.codewars.com/kata/5506b230a11c0aeab3000c1f/kotlin
import kotlin.math.log
import kotlin.math.ceil

fun evaporator(content: Double, evap_per_day: Double, threshold: Double): Int {
  return ceil(log(threshold/100.0, content) / log(1.0-evap_per_day/100.0, content)).toInt()
}